src/services/api/FirebaseApi.ts

```diff
@@ -81,42 +81,50 @@ export class FirebaseApi implements ServiceApi {
     return FirebaseApi.i;
   }
 
-  public async getCourseByGradeId(
+  public async getCourseByUserGradeId(
     gradeDocId: string | undefined
   ): Promise<DocumentReference<DocumentData>[]> {
     let courseIds: DocumentReference[] = [];
-    const courses = await this.getAllCourses();
-    if (!!courses && courses.length > 0) {
-      if (gradeDocId === belowGrade1 || gradeDocId === grade1) {
-        courses.forEach((course) => {
-          //here it repeat all courses but adding only g1 and puzzle
-          if (
-            course.grade.id === grade1 ||
-            course.courseCode === COURSES.PUZZLE
-          ) {
-            courseIds.push(doc(this._db, CollectionIds.COURSE, course.docId));
-          }
-        });
+
+    if (gradeDocId) {
+      // Initialize isGrade1 and isGrade2
+      let isGrade1: string | boolean = false;
+      let isGrade2: string | boolean = false;
+
+      // Check if gradeDocId matches any of the specified grades and assign the value to isGrade1 or isGrade2
+      if (gradeDocId === grade1 || gradeDocId === belowGrade1) {
+        isGrade1 = true;
       } else if (
```

Comment:
Use the 'else' condition here. If it's neither grade1 nor grade2, assume grade2.